src/diffusers/models/adapter.py

```diff
@@ -0,0 +1,245 @@
+                            block_mid_channels[i],
+                            block_mid_channels[i],
+                            down=False,
+                            ksize=kerenl_size,
+                            proj_ksize=proj_kerenl_size,
+                            sk=res_block_skip,
+                            use_conv=use_conv,
+                        )
+                    )
+        self.body = nn.ModuleList(self.body)
+
+        if block_mid_channels[0] == block_out_channels[0]:
+            # follow standar adapter schema, using fix kernel size 3 for stem conv layer
+            self.conv_in = nn.Conv2d(channels_in * input_scale_factor**2, block_mid_channels[0], 3, 1, 1)
+        else:
+            # if block_mid_channels[i] < block_out_channels[i](bottleneck downsample block), using light weight adapter schema instead
+            self.conv_in = nn.Conv2d(
+                channels_in * input_scale_factor**2,
+                block_mid_channels[0],
+                proj_kerenl_size,
+                1,
+                proj_kerenl_size // 2,
+            )
+
+    def forward(self, x: torch.Tensor) -> List[torch.Tensor]:
+        r"""
+        Args:
+            x (`torch.Tensor`):
+                (batch, channel, height, width) input images for adapter model, `channel` should equal to
+                `channels_in`.
+        """
+        # unshuffle
+        x = self.unshuffle(x)
+        # extract features
+        features = []
+        x = self.conv_in(x)
+        for i in range(self.num_downsample_blocks):
+            for j in range(self.num_res_blocks):
+                idx = i * self.num_res_blocks + j
+                x = self.body[idx](x)
+            features.append(x)
+
+        return features
+
+
+class MultiAdapter(ModelMixin):
+    r"""
+    MultiAdapter is a wrapper model that contains multiple adapter models and merges their outputs according to
+    user-assigned weighting.
+
+    This model inherits from [`ModelMixin`]. Check the superclass documentation for the generic methods the library
+    implements for all the model (such as downloading or saving, etc.)
+
+    Parameters:
+        adapters (`List[Adapter]`, *optional*, defaults to None):
+            A list of `Adapter` model instances.
+        adapter_weights (`List[float]`, *optional*, defaults to None):
+            List of floats representing the weight which will be multiply to each adapter's output before adding them
+            together.
+    """
+
+    def __init__(self, adapters: List[Adapter], adapter_weights: Optional[List[float]] = None):
+        super(MultiAdapter, self).__init__()
+
+        self.num_adapter = len(adapters)
+        self.adapters = nn.ModuleList(adapters)
+        if adapter_weights is None:
+            self.adapter_weights = nn.Parameter(torch.tensor([1 / self.num_adapter] * self.num_adapter))
+        else:
+            self.adapter_weights = nn.Parameter(torch.tensor(adapter_weights))
```

Comment:
adapter weights shouldn't be parameters as they're not trainable. Let's pass them directly to the forward function :)

Reply:
Fixed, now adpater_weights will come from the pipeline's `adapter_conditioning_scale` argument during inference/forward pass.